### ISSUE TITLE
TextEditor: formattings are getting lost in PF6.2 RC1

### ISSUE THREAD

**Issue by RueKow** [primefaces/primefaces#3170]
## 1) Environment
- PrimeFaces version: 6.2
- Does it work on the newest released PrimeFaces version? Version? No
- Application server + version: Tomcat 8.5
- Affected browsers: all

## 2) Expected behavior

formattings should be persistent after submitting

## 3) Actual behavior

formattings are getting lost

## 4) Steps to reproduce

see forum [https://forum.primefaces.org/viewtopic.php?f=3&t=53836](https://forum.primefaces.org/viewtopic.php?f=3&t=53836)

## 5) Sample XHTML

```
<html xmlns="http://www.w3.org/1999/xhtml"
	xmlns:h="http://java.sun.com/jsf/html"
	xmlns:f="http://java.sun.com/jsf/core"
	xmlns:ui="http://java.sun.com/jsf/facelets"
	xmlns:p="http://primefaces.org/ui">

    <h:head>
    </h:head>

    <h:body>
		<h:form>
			<p:textEditor widgetVar="editor" value="#{testController.text}" height="300">
			</p:textEditor>
			<p:commandButton value="Submit"
				process="@form"
				update="@form" />
		</h:form>
    </h:body>

</html>
```
## 6) Sample bean

```
@ManagedBean
@SessionScoped
public class testController implements Serializable
{
	private String text;

	public String getText()
	{
		return text;
	}

	public void setText(String text)
	{
		this.text = text;
	}

}
```

**Comment by tandraschko** [comment]
@melloware @jxmai could you check that?

**Comment by melloware** [comment]
Found this on their GitHub...

https://github.com/quilljs/quill/issues/903

**Comment by melloware** [comment]
Actually looking more in the PF Widget it does this code to determine HTML or not.  My guess is the format has changed and this is no longer properly determining whether its HTML or not.

```javascript
getEditorValue: function() {
        var html = $(this.editorContainer[0]).find('p').text();
        var value = (html == '<p><br></p>') ? '' : html;

        return value;
    },
```

**Comment by jxmai** [comment]
I could replicate the bug and I will raise a patch shortly.

```js
var html = $(this.editorContainer[0]).find('p').text();
```

should be changed to 

```js
var html = $(this.editorContainer[0]).find('p')[0].innerHTML;
```

the result:

![image](https://user-images.githubusercontent.com/10467831/35106592-3cea0b36-fc34-11e7-929b-7fbae1231cbe.png)

**Comment by melloware** [comment]
Nice teamwork @jxmai !

**Comment by jxmai** [comment]
No problem :smile: . As a background: I did a fix in the past to get rid of an extra p tag (https://github.com/primefaces/primefaces/issues/2907) that accidentally introduced this regression :sweat_smile: 

The current fix should be able to address both #2907  and #3170
